doc/api/esm.md

```diff
@@ -742,6 +742,62 @@ different [realm](https://tc39.es/ecma262/#realm). The hooks thread may be
 terminated by the main thread at any time, so do not depend on asynchronous
 operations (like `console.log`) to complete.
 
+#### `initialize()`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> The loaders API is being redesigned. This hook may disappear or its
+> signature may change. Do not rely on the API described below.
+
+* `data` {any} The data from `register(loader, import.meta.url, { data })`
+* Returns: {any} The data to be returned to the caller of `register`
+
+The `initialize` hook provides a way to define a custom function that runs
+in the loader's thread when the loader is initialized. This hook can send
+and receive data from a `register` invocation, including ports and other
+transferrable objects.
```

Comment:
```suggestion
transferrable objects. The return value of `initialize` must be something that
can be posted as a message between threads, like the input to
[port.postMessage][].
```

With footnote to https://nodejs.org/api/worker_threads.html#portpostmessagevalue-transferlist. (I’m assuming that this addition is correct?)

Reply:
Yes. `data` as a parameter can be ports and other fun things, but right now the return value has no such magic, unless we wish to change the signature to `return {data, transferList}` or some such.